.github/workflows/build-reactjs.yaml

```diff
@@ -104,7 +104,27 @@ jobs:
 
       - name: Lint check
         run: yarn run prettier:check
-
+  aggregate-coverage:
+    needs: [ tests ]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Download coverage artifacts
+        uses: actions/download-artifact@v4
+        with:
+          pattern: coverage-*
+          path: coverage/
+      - name: Validate coverage artifacts
+        run: |
+          if [ ! -d "coverage" ] || [ -z "$(ls -A coverage)" ]; then
+            echo "No coverage artifacts found. Skipping coverage merge step."
+            exit 1
+          fi
+      - name: Merge and Generate Coverage Report
+        run: |
+          mkdir -p .nyc_output
+          npx nyc merge coverage .nyc_output/out.json
```

Comment:
Sonar understands the lcov format please read the details how Sonar works [here](https://docs.sonarsource.com/sonarqube-server/9.8/analyzing-source-code/test-coverage/javascript-typescript-test-coverage/). 

Please merge the coverage into a lcov format and make it available for the build step. 

Github Actions steps are executed on the same runner, so they see the file from other steps. Sonar has the default `./coverage/lcov.info` path so you can generate the aggregated lcov.info file there.